### PR TITLE
Add rubocop

### PR DESCRIPTION
**Pull request by jinjagit**
First steps in addressing #20  
  
Next step (perhaps for another PR): Integrate with CI - which will need some setup of secrets on this repo. (specifically: `secrets.GITHUB_TOKEN`)

### CODE REVIEW

**Comment by jinjagit** [comment]
**Note:** I have, so far, followed much of the content found in [Ruby on Rails project RuboCop setup with RSpec](https://hixonrails.com/ruby-on-rails-tutorials/ruby-on-rails-project-rubocop-setup-with-rspec/) (just to get going on this WIP and start finding my way around this set of tools). I am aware this covers more ground than just linting and also includes performance and testing tools.

File: app/controllers/decidim/liquidvoting/delegations_controller.rb [hunk 1]
```diff
@@ -11,11 +11,11 @@ def create
           delegator_email: params[:delegator_email],
           delegate_email: params[:delegate_email]
         )
-   
+
         session[:delegated_to] = params[:delegate_email]
         flash[:notice] =
           "Delegated support to #{Decidim::User.find_by(email: session[:delegated_to]).name}."
-      rescue Exception => e
+      rescue StandardError => e
```
Comment (jinjagit) [inline app/controllers/decidim/liquidvoting/delegations_controller.rb:18]: Need for experienced eyes: Rubocop advised this change (2 * in this file, and 2 * in `votes_controller.rb`).  
  
Please check this change is OK. Thanks.

File: lib/decidim/liquidvoting/client.rb [hunk 2]
```diff
@@ -207,32 +219,27 @@ def self.delete_delegation(proposal_url:, delegator_email:, delegate_email:)
...
 
       # return exactly one delegation from delegator_email for proposal_url, or nil
       def self.delegation_for(delegator_email, proposal_url)
         # this is a hack until we can properly query a subset of delegations
-        delegations = self.delegations().
-          select {|d| d.delegator.email == delegator_email && d.proposal_url == proposal_url}.
-          first   # returns nil if list is empty
-      end 
-
-      private
```
Comment (jinjagit) [inline lib/decidim/liquidvoting/client.rb:228]: Need for experienced eyes: Rubocop advised that this use of `private` was useless, so I deleted it.

Please check this change is OK. Thanks.

**Comment by jinjagit** [comment]
So, I have 'fixed' about 60+ 'offenses':  
  
rubocop_says_no_offences
  
Please look over the changes to help ensure I have not made anything worse / broken with my 'fixes'. Thanks!  
I have highlighted 2 particular 'fixes' I most unsure about with specific comments.

File: app/controllers/decidim/liquidvoting/delegations_controller.rb (hunk 1 above)
Comment (oliverbarnes) [reply app/controllers/decidim/liquidvoting/delegations_controller.rb:18]: this is fine 👍

File: app/views/decidim/proposals/proposals/liquid_voting/_session.html.erb [hunk 3]
```diff
@@ -5,8 +5,8 @@
     <% d = Decidim::Liquidvoting::Client.delegation_for(current_user.email, proposal_url(proposal)) %>
     <% session[:delegated_to] = d.delegate.email unless d.nil? %>
   <% end -%>
-  <!-- A blank session[:delegated_to] means we haven't checked yet;
-      let's make it explicitly the delegate email if such a delegation exists -->
+  <!-- A blank session[:voted] means we haven't checked yet;
```
Comment (oliverbarnes) [inline app/views/decidim/proposals/proposals/liquid_voting/_session.html.erb:8]: rubocop caught this? 🤔

File: decidim-liquidvoting.gemspec [hunk 4]
```diff
@@ -10,7 +10,7 @@ Gem::Specification.new do |s|
   s.email = ["oliverbarnes@hey.com"]
   s.license = "AGPL-3.0"
   s.homepage = "https://github.com/liquidvotingio/decidim-liquidvoting"
-  s.required_ruby_version = ">= 2.5"
+  s.required_ruby_version = ">= 2.6"
```
Comment (oliverbarnes) [inline decidim-liquidvoting.gemspec:13]: rubocop caught this?

File: lib/decidim/liquidvoting/client.rb (hunk 2 above)
Comment (oliverbarnes) [reply lib/decidim/liquidvoting/client.rb:228]: Did it say _why_ it's useless?

**Review by oliverbarnes — CHANGES_REQUESTED** [review]
Most changes look good to me, I see a sensible improvement in code quality.

Just a few comments and this should be good to merge

File: app/views/decidim/proposals/proposals/liquid_voting/_session.html.erb (hunk 3 above)
Comment (jinjagit) [reply app/views/decidim/proposals/proposals/liquid_voting/_session.html.erb:8]: No. I caught this. Not strictly the focus of this PR, but also just a comment, so not changing any functionality and was clearly wrong - was my thinking.

File: decidim-liquidvoting.gemspec (hunk 4 above)
Comment (jinjagit) [reply decidim-liquidvoting.gemspec:13]: Rubocop was using the wrong ruby version to check for 'offenses' and it turns out it sets the version to use from that specified in this file.

**Comment by jinjagit** [comment]
@Oliver "Did it say why it's useless?"
  
Yes, but I forgot to copy the message.  
But I remember it said something like: 'Useless private access modifier. Maybe you meant: private_class_method '  
  
I used https://github.com/rubocop-hq/rubocop/issues/6076 as a reference, and decided my best guess was it just doesn't need to be private. But I also flagged it up, as I really don't know enough to decide.

**Review by oliverbarnes — APPROVED** [review]

**Pull request merged**